File: Scala/src/main/scala/Golem/Golem.scala
package Golem

class Golem(__hp: Int = 300, __defense: Int = 80, __attack: Int = 50) {
  private var _hp: Int = __hp
  private val _defense: Int = __defense
  private val _attack: Int = __attack

  def hp: Int = _hp
  def hp_= (delta: Int): Unit = {
    _hp = delta
  }

  def defense: Int = _defense

  def attack: Int = _attack
}

class GolemEx(__hp: Int, __defense: Int, __attack: Int) {
  private var _hp: Int = __hp
  private val _defense: Int = __defense
  private val _attack: Int = __attack
  private val GOLEM_HIT: Int = 70
  private val GOLEM_FLEE: Int = 20

  def Hp: Int = _hp
  def Hp_= (delta: Int): Unit = {
    _hp = delta
  }

  def Defense: Int = _defense
  def Attack: Int = _attack
  def Hit: Int = GOLEM_HIT
  def Flee: Int = GOLEM_FLEE
}

File: rust/src/shared/structs/golem.rs
#[derive(Copy, Clone, Debug)]
pub struct Golem {
    pub hp: i32,
    pub defense: i32,
    pub attack: i32,
}

impl Golem {
    pub fn new(hp: i32) -> Self {
        Golem {
            hp,
            defense: 80,
            attack: 50,
        }
    }
}

impl Default for Golem {
    fn default() -> Self {
        Golem::new(300)
    }
}

pub struct GolemEx {
    pub hp: i32,
    pub defense: i32,
    pub attack: i32,
    pub hit: i32,
    pub flee: i32,
}
impl GolemEx {
    const GOLEM_HIT: i32 = 70;
    const GOLEM_FLEE: i32 = 20;

    pub fn new(hp: i32, defense: i32, attack: i32) -> Self {
        GolemEx {
            hp,
            defense,
            attack,
            hit: GolemEx::GOLEM_HIT,
            flee: GolemEx::GOLEM_FLEE,
        }
    }
}
